core/src/main/scala/org/apache/spark/scheduler/cluster/ExecutorData.scala

```diff
@@ -27,6 +29,8 @@ import org.apache.spark.rpc.{RpcAddress, RpcEndpointRef}
  * @param executorHost The hostname that this executor is running on
  * @param freeCores  The current number of cores available for work on the executor
  * @param totalCores The total number of cores available to the executor
+ * @param totalResources The information of all resources on the executor
```

Comment:
It seems `totalResources` is only used in test. Shall we remove it and rename `availableResources` to `resourceInfo`? It carries all the info we need.

Reply:
we can remove it now if you want, but it will be needed for the UI work but we can add it back in there if you want

Reply:
`resourceInfo` is a superset, and extra info there is also needed by UI

Reply:
Maybe we should move `resourceInfo` into `ExecutorInfo` ? Or we shall do it later when we consider the UI work for extra resources?

Reply:
I would say we try to get this one merged and we can do UI separate and any changes needed there can be done there.  Just remove totalResources for now

Reply:
Sure！

Reply:
removed